src/python/pants/engine/legacy/structs.py

```diff
@@ -30,6 +30,10 @@ class TargetAdaptor(StructWithDeps):
   Extends StructWithDeps to add a `dependencies` field marked Addressable.
   """
 
+  @property
+  def address(self) -> BuildFileAddress:
+    return cast(BuildFileAddress, super().address)
```

Comment:
This is for a more precise type hint. The superclass `Struct` returns an `Optional[BuildFIleAddress]`, whereas here we now that it is always present so can remove the `Optional` part.